### DISCUSSION TITLE
Custom / unique files names when saving pictures from UI

### DISCUSSION THREAD

**Discussion by HonZuna** [lllyasviel/Fooocus#1798] [Ideas]
Hi, if I decide to download the generated image from UI using (right mouse -> save image as) or i just drag it to the desktop/folder with the mouse. The file is always named **image.png** although it has a unique name in the "outputs" folder, e.g. "**2024-01-07_XX-XX-XX_XX**". So a popup always pops up that the same named file is already there.

**A1111 has the same file ID in both cases (saving as or output folder), which makes it easy to drag/save to folders etc.**

I assume that's not even the intention, what do you think would be worth modifying?

Thank you !

**Comment by poor7** [comment]
This bug.

**Reply by HonZuna** [reply]
You are right i guess.

**Comment by mashb1t** [comment]
~~This is a bug but i'm not able to reproduce it on either Mac, Windows or Colab... Any help would be appreciated!~~
It's an enhancement / feature, see https://github.com/lllyasviel/Fooocus/issues/1421

**Comment by HonZuna** [comment]
> ~This is a bug but i'm not able to reproduce it on either Mac, Windows or Colab... Any help would be appreciated!~ It's an enhancement / feature, see #1421

With all respect, bad feature at best.

**Reply by mashb1t** [reply]
This feature enables returning not only png, but also jpg and webp, which wouldn't be possible without the changes. Furthermore, without this feature meta data handling wouldn't be possible in the browser.
See https://github.com/lllyasviel/Fooocus/pull/1863 + https://github.com/lllyasviel/Fooocus/pull/1940

How would you have solved this?
